prometheus_es_exporter/nodes_stats_parser.py

```diff
@@ -37,7 +46,16 @@ def parse_block(block, metric=[], labels={}):
                     else:
                         singular_key = key
                     for n_key, n_value in value.items():
-                        result.extend(parse_block(n_value, metric=metric + [key], labels=merge_dicts(labels, {singular_key: [n_key]})))
+                        if isinstance(n_value, list):
+                            if key == "shards":
+                                n_list = [{"shard_id": d} for d in n_value]
+                                for n_value in n_list:
+                                    result.extend(parse_block(n_value, metric=metric + [key], labels=merge_dicts(labels, {singular_key: [n_key]})))
+                        else:
+                            if key in excluded_metric_keys:
```

Comment:
Why is the key being added to the result when it's supposed to be `excluded`? This seems semantically wrong.

Reply:
The `excluded_metric_keys` specifies which keys shouldn't appear in the metric name, not that they shouldn't be added to the result. 

Since the metric name is built by iterating over the dict keys, I needed a way to exclude some specific keys (e.g. `shard_id`, in our case). That's why the parameter `metric` is set to itself, instead of `metric + [key]`.

If it seems confusing to you, would a comment that explains what it's used for be enough?

Reply:
Yep, a comment would be enough for me. Thank you.